src/compiler/abstract_compiler.nit

```diff
@@ -3632,9 +3698,39 @@ redef class ASuperstringExpr
 		# Fast join the native string to get the result
 		var res = v.send(v.get_property("native_to_s", a.mtype), [a])
 
+		if is_re then
+			if res == null then abort
+			res = v.send(v.get_property("to_re", res.mtype), [res])
+			if res == null then
+				print "Cannot call property `to_re` on {self}"
+				abort
+			end
+			for i in suffix.chars do
+				if i == 'i' then
+					var prop = v.get_property("ignore_case=", res.mtype)
+					v.send(prop, [res, v.bool_instance(true)])
+					continue
+				end
+				if i == 'm' then
+					var prop = v.get_property("newline=", res.mtype)
+					v.send(prop, [res, v.bool_instance(true)])
+					continue
+				end
+				if i == 'b' then
+					var prop = v.get_property("extended=", res.mtype)
+					v.send(prop, [res, v.bool_instance(false)])
+					continue
+				end
+				# Should not happen, this needs to be updated
+				# along with the addition of new suffixes
+				abort
+			end
```

Comment:
That's a big duplicated block of code here... is there a difference to the implementation on AStringExpr?